src/sqlfluff/rules/L067.py

```diff
@@ -0,0 +1,156 @@
+"""Implementation of Rule L067."""
+from typing import Optional
+
+from sqlfluff.core.parser import NewlineSegment
+
+from sqlfluff.core.rules.base import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+from sqlfluff.core.rules.doc_decorators import document_fix_compatible, document_groups
+from sqlfluff.utils.functional import sp, FunctionalContext
+
+
+@document_groups
+@document_fix_compatible
+class Rule_L067(BaseRule):
+    """A subquery inside parentheses should start on a new line.
+
+    For brackets, if there are any subqueries inside, the subqueries
+    should begin on the next line after the starting bracket, and the
+    ending bracket should go on the next line after the subquery.
+
+    **Anti-pattern**
+
+    Parentheses on the line with a subquery.
+
+    .. code-block:: sql
+        :force:
+
+        SELECT A.SOMETHING
+        FROM (SELECT
+                B,
+                C
+            FROM D GROUP BY 1) AS A
+
+    **Best practice**
+
+    Opening bracket and ending bracket take a new line out of the subquery.
+
+    .. code-block:: sql
+        :force:
+
+        SELECT A.SOMETHING
+        FROM (
+            SELECT
+                B,
+                C
+            FROM D GROUP BY 1
+        ) AS A
+
+    """
+
+    groups = ("all",)
+    crawl_behaviour = SegmentSeekerCrawler({"bracketed"})
+
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
```

Comment:
This rule is intended to work in conjunction with your other PR, which adds rule L068. I wonder if you should combine the two PRs, because we're going to want tests that test the interaction of the two rules.